### PR TITLE
[Ahuna] Confirmation modals when changing sharing / deleting

### LINKED ISSUE #3272: Update sharing changes modal copy
As per [here](https://www.figma.com/file/QOxHwppG64GtPocpDhS6Y7/Product?type=design&node-id=4050-22850&mode=design&t=675WBPvIgubbwQiv-4)

Related to #3267 

-----
Full [figma](https://www.figma.com/file/QOxHwppG64GtPocpDhS6Y7/Product?type=design&node-id=4050-22251&mode=design&t=Dlhq25QyeMahW2OI-0) (source of truth)

Full [issue](https://github.com/dust-tt/decisions/issues/147#issue-2081773564)

Even clearer specs via the [prototype](https://www.figma.com/proto/QOxHwppG64GtPocpDhS6Y7/Product?type=design&node-id=4050-22252&t=PrquaF04miEgyEMq-0&scaling=min-zoom&page-id=4050%3A22251&starting-point-node-id=4050%3A22252&show-proto-sidebar=1)

### PR DESCRIPTION
**Pull request by philipperolet**
## Description
Fixes  #3272 (see issue for details)

Will follow-up with 2 PRs:
- Currently, snap closes. will fix in following PR with a new sparkle setting
- hover on chip text is mouse pointer (pointer style should be at chip level not text, warrants a sparkle change, TBD with @Duncid) 10'

Screenshot from 2024-02-01 17-16-54
Screenshot from 2024-02-01 17-16-33
Screenshot from 2024-02-01 17-16-21

## Risk
Only UX & copy so risks are limited

### CODE REVIEW

File: front/lib/swr.ts [hunk 1]
```diff
@@ -537,14 +537,16 @@ export function useAgentConfiguration({
   agentConfigurationId,
 }: {
   workspaceId: string;
-  agentConfigurationId: string;
+  agentConfigurationId: string | null;
```
Comment (flvndvd) [inline front/lib/swr.ts:540]: This is the hack that you were explaining IRL, right?

File: front/components/assistant_builder/TeamSharingSection.tsx [hunk 2]
```diff
@@ -90,29 +169,80 @@ export function TeamSharingSection({
...
+                          agentUsage.agentUsage.userCount <= 1))
+                    ) {
+                      console.log("no modal");
+                      setNewScope(
+                        entryScope as Exclude<AgentConfigurationScope, "global">
+                      );
+                      return;
+                    }
+                    // in all other cases, show modal
+                    console.log("show modal");
```
Comment (flvndvd) [inline front/components/assistant_builder/TeamSharingSection.tsx:194]: ✂️

File: front/components/assistant_builder/TeamSharingSection.tsx [hunk 3]
```diff
@@ -90,29 +169,80 @@ export function TeamSharingSection({
...
+                      // selection back to initial state
+                      entryScope === initialScope ||
+                      // change to personal or company, but the only user of the
+                      // assistant is the user changing the scope
+                      ((entryScope === "private" || entryScope === "company") &&
+                        assistantInMyList &&
+                        (!agentUsage.agentUsage ||
+                          agentUsage.agentUsage.userCount <= 1))
+                    ) {
+                      console.log("no modal");
```
Comment (flvndvd) [inline front/components/assistant_builder/TeamSharingSection.tsx:187]: ✂️

File: front/components/assistant_builder/TeamSharingSection.tsx [hunk 4]
```diff
@@ -9,74 +10,152 @@ import {
...
+  };
   return (
     <div className="flex flex-col gap-3">
       <div className="text-lg font-bold text-element-900">Sharing</div>
       <div>
+        {requestNewScope && (
+          <ScopeChangeModal
+            show={requestNewScope !== null}
+            confirmationModalData={
+              requestNewScope
```
Comment (flvndvd) [inline front/components/assistant_builder/TeamSharingSection.tsx:133]: Can we extract this part and avoid nested ternaries?

File: front/components/assistant_builder/TeamSharingSection.tsx [hunk 5]
```diff
@@ -9,74 +10,152 @@ import {
...
+      },
     },
     workspace: {
       label: "Company Assistant",
       color: "amber",
       icon: PlanetIcon,
       text: "Activated by default for all members of the workspace.",
+      confirmationModalData: {
+        title: "Moving to Company Assistants",
+        text: "Moving the assistant to Company Assistants will make the assistant editable only by Admins and Builders and add it the “My Assistants” list of all the workspace.",
```
Comment (flvndvd) [inline front/components/assistant_builder/TeamSharingSection.tsx:96]: Copy seems wrong here.

**Review by Duncid — APPROVED** [review]

File: front/components/assistant_builder/TeamSharingSection.tsx (hunk 4 above)
Comment (philipperolet) [reply front/components/assistant_builder/TeamSharingSection.tsx:133]: we can; I agree it's not optimal;  but other options don't seem very good too (if requires a let + null check etc). Do you see a way of writing this well & concisely?

File: front/components/assistant_builder/TeamSharingSection.tsx (hunk 5 above)
Comment (philipperolet) [reply front/components/assistant_builder/TeamSharingSection.tsx:96]: It's the figma copy; but yeah will do a minor tweak

**Review by flvndvd — APPROVED** [review]

File: front/components/assistant_builder/TeamSharingSection.tsx (hunk 5 above)
Comment (Duncid) [reply front/components/assistant_builder/TeamSharingSection.tsx]: I'll do a pass on copy in a new PR.

**Pull request merged**